src/main/java/com/capstone/producer/TfrHandler.java

```diff
@@ -0,0 +1,179 @@
+package com.capstone.producer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.capstone.producer.kafka.KafkaProducer;
+
+@Service
+public class TfrHandler {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TfrHandler.class);
+    /**
+     * Keeps track of all TFRs Injected
+     */
+    private static Map<String, String[]> receivedNotams;
```

Comment:
Why is it necessary to keep track of all notam messages received?

Reply:
Notams can be spread across multiple message parts that are injected as multiple messages. This allows us to keep all the parts until we have the complete message.

Reply:
Cool you link the docs for the notam message. I feel like I don't really understand how they work and would like to learn more about it.

Reply:
https://github.com/capstone-team-night-watch/utilities/blob/main/currentTFRs.txt
https://tfr.faa.gov/tfr2/list.html
Unfortunately there was no documentation I could find on parsing NOTAM TFRs, everything I made was based on evaluating multiple at the same time.